test/sqllogictest/string.slt

```diff
@@ -939,6 +939,128 @@ SELECT position(e'\u0903\u0951' IN e'\u0930\u0903\u0951')
 ----
 2
 
+### left ###
+statement ok
+CREATE TABLE lefttest (strcol char(15), vccol varchar(15), smicol smallint, intcol int)
+
+statement ok
+INSERT INTO lefttest VALUES ('Mg', 'Mn', 1, -1), ('magnesium', 'manganese', 3, NULL),
+    (NULL, NULL, 0, 0), ('24.31', '54.94', 3, -3), ('长久不见', '爱不释手', NULL, 3),
+    ('', '', -1, 2)
+
+# invalid input
+statement error
+SELECT left(42)
+
+statement error
+SELECT left('str')
+
+statement error
+SELECT left(42, 'str')
+
+statement error
+SELECT left('str', 42, 17)
+
+# standard tests
+
+#TODO: materialize#589 select left(strcol, foo) from lefttest
+
+# edge case
+query T rowsort
+SELECT left(vccol, 0) FROM lefttest
+----
+(empty)
+(empty)
+NULL
+(empty)
+(empty)
+(empty)
+
+# from the back
+query T rowsort
+SELECT left(vccol, 2) FROM lefttest
+----
+Mn
+ma
+NULL
+54
+爱不
+(empty)
+
+# from the front
+query T rowsort
+SELECT left(vccol, -2) FROM lefttest
+----
+(empty)
+mangane
+NULL
+54.
+爱不
+(empty)
+
+query TT rowsort
+SELECT left(vccol, smicol), left(vccol, intcol) FROM lefttest
+----
+M   M
+man NULL
+NULL    NULL
+54.    54
+NULL    爱不释
+(empty) (empty)
+
+
+# NULL inputs
+query T
+SELECT left(NULL, 99)
+----
+NULL
+
+query T
+SELECT left('str', NULL)
+----
+NULL
+
+# combining characters
+
+query T
+SELECT left('रः॑', 0)
+----
+(empty)
+
+query T
+SELECT left('रः॑', 1)
+----
+ र
+
+query T
+SELECT left('रः॑', -1)
+----
+रः
+
+# this is exactly the same as above, but using unicode escapes
+query T
+SELECT left(e'\u0930\u0903\u0951', -1)
+----
+रः
+
+# integer edge cases
```

Comment:
Did you check whether these match what Postgres would be doing?

Reply:
Yep, same result.